add version condition to route even without any routing annotations

dmaicher opened the pull request:
Fixes https://github.com/FriendsOfSymfony/FOSRestBundle/issues/1386 and https://github.com/FriendsOfSymfony/FOSRestBundle/issues/1436

dmaicher commented:
Travis CI build fail is unrelated

Ener-Getick commented:
LGTM :+1:

Ener-Getick commented on Routing/Loader/Reader/RestActionReader.php:
```diff
@@ -333,39 +333,41 @@ public function read(RestRouteCollection $collection, \ReflectionMethod $method,
+            return $conditionTwo;
+        }
+
+        if ($conditionTwo === null) {
```
You should use the yoda notation here:
```php
null === $conditionTwo
```
same thing for the condition above

dmaicher replied:
I personally prefer to not use yoda style but sure I can change it :+1:

Ener-Getick commented on Routing/Loader/Reader/RestActionReader.php:
```diff
@@ -333,39 +333,41 @@ public function read(RestRouteCollection $collection, \ReflectionMethod $method,
+            return;
+        }
+
+        return sprintf("request.attributes.get('version') in ['%s']", implode("','", $this->versions));
```
I'd prefer to add a space after the comma and to use double quotes:
```php
implode('", "', $this->versions)
```

dmaicher replied:
Agree with the space after the comma to improve readability. But why the double quotes? Any reason apart from aesthetics? Just curious :)

Ener-Getick commented on Routing/Loader/Reader/RestActionReader.php:
```diff
@@ -333,39 +333,41 @@ public function read(RestRouteCollection $collection, \ReflectionMethod $method,
             // add route to collection
             $route = new Route(
-                $path, $defaults, $requirements, $options, $host, [], $methods, $condition
+                $path, $defaults, $requirements, $options, $host, [], $methods, $versionCondition
```
I'm not sure we should rename this variable, this looks a bit confusing with the one which keeps the same name.

dmaicher replied:
Well at this point we are only adding the version condition which was defined in line 295. So I think its actually less confusing like this?

dmaicher replied:
Maybe I can rename `$condition` in line 319 to `$combinedConditions` or similar?

dmaicher commented:
@Ener-Getick addressed some of your comments. Let me know if something else should be fixed. I kept the single quotes for the expression as they are used like this all over the code ;)

Ener-Getick commented:
LGTM :+1:

Ener-Getick commented:
Thank you @dmaicher

dmaicher commented:
No problem :+1:

stloyd commented on Tests/Routing/Loader/RestRouteLoaderTest.php:
```diff
@@ -173,7 +173,7 @@ public function testAnnotatedVersionUserFixture()
 
         $this->assertTrue($collection instanceof RestRouteCollection);
-        $this->assertEquals(2, count($collection->all()));
+        $this->assertEquals(3, count($collection->all()));
```
`assertCount()` should be used instead.

dmaicher replied:
true could have fixed that :+1:
